homeassistant/components/roborock/select.py

```diff
@@ -79,6 +79,12 @@ async def async_setup_entry(
         )
         is not None
     )
+    async_add_entities(
+        RoborockCurrentMapSelectEntity(
+            f"selected_map_{slugify(coordinator.duid)}", coordinator
```

Comment:
This code appears quite a bit
```
binary_sensor.py:from homeassistant.util import slugify
binary_sensor.py:            f"{description.key}_{slugify(coordinator.duid)}",
button.py:from homeassistant.util import slugify
button.py:            f"{entity_description.key}_{slugify(coordinator.duid)}",
image.py:from homeassistant.util import slugify
image.py:                f"{slugify(coord.duid)}_map_{map_info.name}",
number.py:from homeassistant.util import slugify
number.py:                    f"{description.key}_{slugify(coordinator.duid)}",
select.py:from homeassistant.util import slugify
select.py:            f"selected_map_{slugify(coordinator.duid)}", coordinator
select.py:            f"{entity_description.key}_{slugify(coordinator.duid)}",
sensor.py:from homeassistant.util import slugify
sensor.py:            f"{description.key}_{slugify(coordinator.duid)}",
sensor.py:        super().__init__(f"{description.key}_{slugify(coordinator.duid)}", coordinator)
switch.py:from homeassistant.util import slugify
switch.py:                    f"{description.key}_{slugify(coordinator.duid)}",
time.py:from homeassistant.util import slugify
time.py:                    f"{description.key}_{slugify(coordinator.duid)}",
vacuum.py:from homeassistant.util import slugify
vacuum.py:            slugify(coordinator.duid),
```

Maybe make `coordinator.duid_slug`

Reply:
Also since it never changes you could make it a cached_property so it doesn't have to execute every time

```diff
diff --git a/homeassistant/components/roborock/coordinator.py b/homeassistant/components/roborock/coordinator.py
index 430e2815a7b..c6746be165e 100644
--- a/homeassistant/components/roborock/coordinator.py
+++ b/homeassistant/components/roborock/coordinator.py
@@ -5,6 +5,7 @@ from __future__ import annotations
 import asyncio
 from datetime import timedelta
 import logging
+from functools import cached_property
 
 from roborock import HomeDataRoom
 from roborock.containers import DeviceData, HomeDataDevice, HomeDataProduct, NetworkInfo
@@ -142,11 +143,15 @@ class RoborockDataUpdateCoordinator(DataUpdateCoordinator[DeviceProp]):
                         self._home_data_rooms.get(room.iot_id, "Unknown")
                     )
 
-    @property
+    @cached_property
     def duid(self) -> str:
         """Get the unique id of the device as specified by Roborock."""
         return self.roborock_device_info.device.duid
 
+    @cached_property
+    def duid_slug(self) -> str:
+        """Get the slug of the duid."""
+        return slugify(self.duid)
 
 class RoborockDataUpdateCoordinatorA01(
     DataUpdateCoordinator[
```